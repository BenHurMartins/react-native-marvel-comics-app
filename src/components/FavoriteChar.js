import React from 'react';
import {TouchableOpacity, Image, Text, View} from 'react-native';
import {useNavigation} from '@react-navigation/native';

const FavoriteChar = ({favoriteChar}) => {
  const {name} = favoriteChar;
  const navigation = useNavigation();

  const exploreComics = () => {};
  console.log(
    `${favoriteChar.thumbnail?.path.toString()}/portrait_uncanny.jpg`,
  );
  return (
    <View style={{width: '100%'}}>
      <Image
        source={{
          uri: `${favoriteChar.thumbnail?.path.toString()}/portrait_uncanny.jpg`.replace(
            'http://',
            'https://',
          ),
        }}
        style={{height: 150, width: 100, resizeMode: 'contain'}}
        resizeMode={'contain'}
        height={120}
        width={120}
      />
      <TouchableOpacity
        onPress={() => navigation.navigate('ListComics', {favoriteChar})}>
        <Text>Explore Comics</Text>
      </TouchableOpacity>
      <Text>{name}</Text>
    </View>
  );
};

export default FavoriteChar;
