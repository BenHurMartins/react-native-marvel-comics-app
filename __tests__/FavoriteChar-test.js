/**
 * @format
 */

import 'react-native';
import React from 'react';
import {render, act} from '@testing-library/react-native';
import FavoriteChar from '../src/components/FavoriteChar';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const favoriteChar = {
  name: 'Super Ben',
  description: 'A react native developer born in 1990;',
};

describe('Redenring tests for ListComics', () => {
  it('FavoriteChar renders correctly', () => {
    renderer.create(<FavoriteChar favoriteChar={favoriteChar} />);
  });
  it('FavoriteChar snapshot test', () => {
    const tree = renderer
      .create(<FavoriteChar favoriteChar={favoriteChar} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Should return render the value passed', () => {
    const {getByText} = render(<FavoriteChar favoriteChar={favoriteChar} />);
    act(() => {
      expect(getByText(favoriteChar.name)).not.toBe(null);
    });
  });
});
