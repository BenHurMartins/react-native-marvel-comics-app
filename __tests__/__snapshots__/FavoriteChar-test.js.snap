// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redenring tests for ListComics FavoriteChar snapshot test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginVertical": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#EDF4ED",
        "borderRadius": 13,
        "marginBottom": 10,
        "width": 710,
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "#EDF4ED",
          "color": "#000000",
          "fontSize": 17,
          "fontWeight": "bold",
          "marginHorizontal": 10,
          "marginTop": 10,
        }
      }
    >
      Super Ben
    </Text>
    <Text
      style={
        Object {
          "backgroundColor": "#EDF4ED",
          "color": "#000000",
          "fontSize": 14,
          "fontWeight": "400",
          "marginBottom": 10,
          "marginHorizontal": 10,
          "marginTop": 7,
          "textAlign": "justify",
        }
      }
    >
      A react native developer born in 1990;
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "borderBottomLeftRadius": 0,
        "borderBottomRightRadius": 0,
        "borderColor": "#e62429",
        "borderWidth": 1,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "width": 710,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 20,
          },
          undefined,
        ]
      }
    >
      Explore Comics
    </Text>
  </View>
</View>
`;
